sdcm/utils/validators/iotune.py

```diff
@@ -0,0 +1,102 @@
+        self.preset_io_properties: IOProperties = {}
+        self.results = {
+            "deviation": {},
+            "live": {},
+            "limits":  {},
+        }
+
+    def validate(self):
+        self._run_io_tune()
+        self._prepare_results_table()
+        self._format_results_to_console()
+
+    def _read_io_properties(self, io_props_path="/etc/scylla.d/io_properties.yaml") -> IOProperties:
+        with remote_file(self.node.remoter, io_props_path) as f:
+            return yaml.safe_load(f)
+
+    def _run_io_tune(self, temp_props_path="/tmp/io_properties.yaml") -> IOProperties:
+        self.node.remoter.sudo(f"iotune  --evaluation-directory /var/lib/scylla --properties-file {temp_props_path}")
+        self.node_io_properties = self._read_io_properties(temp_props_path)
+        self.preset_io_properties = self._read_io_properties()
+
+        preset_disk = next(iter(self.preset_io_properties.get("disks", [])), None)
+        if not preset_disk:
+            LOGGER.error("Unable to continue - node should have io_properties.yaml, but it doesn't.")
+            TestFrameworkEvent(source="send_iotune_results_to_argus",
+                               message="Unable to continue - node should have io_properties.yaml, but it doesn't.",
+                               severity=Severity.ERROR).publish()
+            raise IOTuneException("Unable to continue - node should have io_properties.yaml, but it doesn't.")
+
+        return self.node_io_properties
+
+    @staticmethod
+    def _bottom_limit(metric_val, threshold=0.15):
+        """
+            Determine disk metric deviation threshold and
+            use that as fixed limit
+        """
+        return metric_val * threshold
+
+    def _prepare_results_table(self):
+        preset_disk = next(iter(self.preset_io_properties.get("disks", [])), None)
+        tested_disk = deepcopy(next(iter(self.node_io_properties.get("disks", []))))
+        tested_mountpoint = tested_disk.pop("mountpoint")
+        if tested_mountpoint != preset_disk["mountpoint"]:
+            LOGGER.error("Disks differ - probably a mistake: %s vs %s",
+                         tested_mountpoint, preset_disk["mountpoint"])
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"Disks differ - probably a mistake: {tested_mountpoint} vs {preset_disk['mountpoint']}",
+                               severity=Severity.ERROR).publish()
+            raise IOTuneException(
+                f"Disks differ - probably a mistake: {tested_mountpoint} vs {preset_disk['mountpoint']}")
+
+        for key, value in tested_disk.items():
+            deviation_val = abs(preset_disk.get(key) - value)
+            self.results["deviation"][key] = deviation_val
+            self.results["live"][key] = value
+            self.results["limits"][key] = self._bottom_limit(preset_disk[key])
+
+    def _format_results_to_console(self):
```

Comment:
why not basing on self.results?

Reply:
Could you elaborate? I'm not sure what's wrong here.

Reply:
Ah, I see now, yes, that could be fixed, before it was because it needed copies of the structures, now the structure it needs is done for it before, will fix.

Reply:
Fixed

Reply:
good,
that got me thinking, maybe instead sending deviation and showing limit in absolute value, how about send abs deviation in % and set fixed limit to 15%?
This way graph/results would be even more readable.

Reply:
Could be done, I could send it as a third table to have both abs and % values.

Reply:
Done